core/src/main/java/org/infinispan/configuration/cache/HashConfiguration.java

```diff
@@ -138,6 +148,10 @@ public float capacityFactor() {
       return capacityFactor.get();
    }
 
+   public KeyPartitionerFactory keyPartitionerFactory() {
+      return keyPartitionerFactory.get();
+   }
+
```

Comment:
I wouldn't make it mutable, because changing it after the cache was started would make keys inaccessible - we'd need to do some rebalancing to handle this without data loss.  TBH I'm not a fan of modules modifying the configuration in general, I believe the configuration should always match what the user set.

In your case, perhaps a better solution would be to register the `KeyPartitionerFactory` (or the `KeyPartitioner`) in the component registry, and allow modules to override it there.